mlflow/deployments/base_plugin.py

```diff
@@ -0,0 +1,31 @@
+import sys
+import abc
+
+
+if sys.version_info >= (3, 4):
```

Comment:
Should we wait for that PR to merge? Or maybe I can take this up in a consecutive PR?

Reply:
Also, It's better to notify you that the metaclass setup is broken in several places in MLFlow source code. It's because there are classes that use python 2.x style metaclass declaration (using `__metaclass__`) which is ignored in python 3.x. So when you migrate those metaclasses to python3x, there is a chance that child classes are going to break

Reply:
Thanks for the heads up, will take a look at the metaclass issue - IMO it's fine to take this up later unless #2534 merges first